src/boimmgpy/model_annotation/model_annotator.py

```diff
@@ -0,0 +1,382 @@
+        backbone = None
+        found = False
+        lipid_class_dict = defaultdict(int)
+        check_annotation_dict={}
+        counter = {}
+        results = {}
+        side_chain = []
+        for match in matches:
+            found = True
+            metabolite_name = metabolite_name.replace(
+                match.string[match.start() : match.end()], ""
+            )
+            side_chain.append(match.string[match.start() : match.end()])
+
+        if found:
+            check_annotation_dict = self.annotation_checker(lipid=metabolite)
+            backbone = re.sub(" *(\([\-a-zA-Z0-9/|, ]*\))", "", metabolite_name)
+            if len(side_chain) != 1:
+                for a in range(len(side_chain) - 1):
+                    backbone = re.sub(" *(\([\-a-zA-Z0-9/|, ]*\))", "", backbone)
+            lipid_class_dict[backbone] += 1
+            self.backbone = backbone
+            counter,results = self.search_lipid_synonyms(metabolite,side_chain)
+            
+
+        sorted_lipid_class_dict = sorted(
+            lipid_class_dict.items(), key=lambda x: x[1], reverse=True
+        )
+        converted_lipid_class_dict = dict(sorted_lipid_class_dict)
+
+        return (
+            converted_lipid_class_dict,
+            check_annotation_dict,
+            counter,
+            results,
+            
+        )
+
+    def annotation_checker(self, lipid):
+        """Method that checks if a given lipid is annotated in the model
+
+        Args:
+            lipid (_type_): Given lipid metabolite from the model
+        """
+        check_annotation_dict = {}
+        annotation = lipid.annotation.keys()
+        annotated = False
+        check_annotation_dict[lipid.id] = annotated
+        if "slm" in annotation or "lipidmaps" in annotation:
+            annotated = True
+            check_annotation_dict[lipid.id] = annotated
+        return check_annotation_dict
+
+    def search_lipid_synonyms(self, metabolite:Metabolite,side_chain:List[str]):
+        """Method that implements the screening in the database for accurate lipid structure.
+        This method calls all StaticMethods to do the screening in the database.
+
+        Args:
+            metabolite (Metabolite): Given Metabolite of the cobra model
+            side_chain (List[str]): List of the side chains referent to the given Metavolite
+
+        Returns:
+            _type_: tuple with two dictionaries, first one for the extend of the lipid classes that the algorithm caugth and second one for the identifiers to do the annotation
+        """
+        session = self.login()
+        results={}
+        counter = defaultdict(int)
+        is_compound = False
+        lipid_id = self.get_synonym_id(session,self.backbone, side_chain)
+        if lipid_id != None and not lipid_id[2]:
+            for backbone in lipid_id[0]:
+                lipid_compound = self.get_coumpound(session,backbone, lipid_id[1], is_compound)
```

Comment:
there's a typo here in the **get_coumpound** method name. replace to **get_compound**